external/fv3fit/tests/training/test_autoencoder.py

```diff
@@ -0,0 +1,174 @@
+import numpy as np
+import xarray as xr
+from typing import Sequence
+from fv3fit.pytorch.cyclegan import AutoencoderHyperparameters, train_autoencoder
+from fv3fit.pytorch.cyclegan.train import TrainingConfig
+from fv3fit.tfdataset import iterable_to_tfdataset
+import collections
+import os
+import fv3fit.pytorch
+import fv3fit
+
+
+def get_tfdataset(nsamples, nbatch, ntime, nx, ny, nz):
```

Comment:
Can this get a docstring with maybe some equations or visual description of what's happening here? (You do this in the test itself below but more would be helpful.) 

The setup of the test data is pretty complicated. Does the test actually need to be this complex to demonstrate that the autoencoder works?

Reply:
Added a description to the function.

Reply:
OK that sounds fine for now then.